presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2295,6 +2341,16 @@ public PhysicalOperation visitSemiJoin(SemiJoinNode node, LocalExecutionPlanCont
             return new PhysicalOperation(operator, outputMappings, context, probeSource);
         }
 
+        private Set<DynamicFilterId> getCoordinatorDynamicFilters(Set<DynamicFilterId> dynamicFilters, PlanNode node, LocalExecutionPlanContext context)
+        {
+            if (isBuildSideReplicated(node) && context.getTaskId().getId() > 0) {
```

Comment:
can we use `getTaskId()` here?  If so, we can drop that parameter

Reply:
I would actually say `== 0` and maybe add a comment that we only collect DF on the first task.... something really clear what is going on